tutorials/compiler-explorer-deployment.md

```diff
@@ -0,0 +1,227 @@
+<!--
+SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+-->
+
+# Tutorial: Compiler Explorer Deployment
+
+Full instructins are available at [compiler-explorer: Adding a new library](https://github.com/compiler-explorer/compiler-explorer/blob/main/docs/AddingALibrary.md). This page is a Beman cheatsheet.
+
+## Integrating a header-only library
+
+### Step by step example: `beman.optional`
+
+1. Clone Compiler Explorer repos:
+
+```bash
+$ git clone https://github.com/compiler-explorer/infra
+$ git clone https://github.com/compiler-explorer/compiler-explorer
+```
+
+2. Update `infra/`:
+
+   2.1. Locate an existing Beman library inside `bin/yaml/libraries.yaml` - e.g. search for `repo: bemanproject/optional`.
+
+   2.2. Add a new entry (keeping Beman entries grouped and sorted):
+
+```yml
+beman_optional:
+  build_type: none
+  check_file: README.md
+  method: nightlyclone
+  repo: bemanproject/optional
+  targets:
+  - main
+  type: github
+```
+
+Notes:
+
+* `build_type: none`: Not required to run a build command (header-only libraries).
+* `check_file: README.md`: The platform checks the hash of this file.
+* `method: nightlyclone`: Do a deployment every day. We don't want and don't need more often deployment.
+* `repo: bemanproject/optional`: Clone and deploy `beman.optional` from <https://github.com/bemanproject/optional>.
+* `target: ... main`: Only deploy main as trunk version.
+* `type: github`: The repo will be always clone with latest version.
+
+  2.3. Deploy local `infra/` instance. For header-only libraries, it's enough to just run:
+
+```bash
+# Create default deployment directory and make sure to set write permissions for your user.
+$ mkdir -p /opt/compiler-explorer
+$ ls -l /opt/                  
+drwxr-xr-x  25 dariusn  wheel   800 Apr  7 21:29 compiler-explorer
+
+$ cd infra/
+$ make ce
+$ bin/ce_install --enable nightly install beman_optional
+...
+ 2025-04-07 23:05:07,137 lib.ce_install  INFO     libraries/c++/nightly/beman_optional main installed OK
+1 packages installed OK, 0 skipped, and 0 failed installation
+```
+
+Notes: Tested on `MacOS 14.6` (aarch64) and `Ubuntu 24.04` (aarch64).
+
+  2.4. On success, you can find the new Beman library deployed on local setup in `/opt/compiler-explorer`. E.g., for `beman.optional` installation should see this tree:
+
+```bash
+$ tree /opt/compiler-explorer/libs/beman_optional
```

Comment:
```suggestion
$ tree --charset=ascii /opt/compiler-explorer/libs/beman_optional
```

we can use this to keep only ascii characters in this file.

Reply:
done